From: Ben Hutchings <ben@decadent.org.uk>
Date: Sun, 8 Jan 2012 00:36:59 +0000
Subject: [PATCH 1/2] block: Restore blk_init_allocated_queue_node() for ABI
 compatibility

This function was removed as redundant in 3.1.8, but it was previously
exported to modules.
---
 block/blk-core.c       |    9 +++++++++
 include/linux/blkdev.h |    3 +++
 2 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/block/blk-core.c b/block/blk-core.c
index 8fc4ae2..9fce4c9 100644
--- a/block/blk-core.c
+++ b/block/blk-core.c
@@ -549,6 +549,15 @@ blk_init_allocated_queue(struct request_queue *q, request_fn_proc *rfn,
 }
 EXPORT_SYMBOL(blk_init_allocated_queue);
 
+struct request_queue *
+blk_init_allocated_queue_node(struct request_queue *q, request_fn_proc *rfn,
+			      spinlock_t *lock, int node_id)
+{
+	/* q->node was already set by blk_alloc_queue_node(); ignore node_id */
+	return blk_init_allocated_queue(q, rfn, lock);
+}
+EXPORT_SYMBOL(blk_init_allocated_queue_node);
+
 int blk_get_queue(struct request_queue *q)
 {
 	if (likely(!test_bit(QUEUE_FLAG_DEAD, &q->queue_flags))) {
diff --git a/include/linux/blkdev.h b/include/linux/blkdev.h
index 5e30b45..7fbaa91 100644
--- a/include/linux/blkdev.h
+++ b/include/linux/blkdev.h
@@ -803,6 +803,9 @@ extern void blk_unprep_request(struct request *);
  */
 extern struct request_queue *blk_init_queue_node(request_fn_proc *rfn,
 					spinlock_t *lock, int node_id);
+extern struct request_queue *blk_init_allocated_queue_node(struct request_queue *,
+							   request_fn_proc *,
+							   spinlock_t *, int node_id);
 extern struct request_queue *blk_init_queue(request_fn_proc *, spinlock_t *);
 extern struct request_queue *blk_init_allocated_queue(struct request_queue *,
 						      request_fn_proc *, spinlock_t *);
-- 
1.7.8.2

